components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/CacheBackedApplicationDAO.java

```diff
@@ -734,4 +807,78 @@ private static void clearAppCacheByInboundKey(ServiceProvider serviceProvider, S
             }
         }
     }
+
+    /**
+     * Clear the trusted app cache for the available platform types of all tenants depending on the
+     * availability of platform specific data.
+     *
+     * @param spTrustedAppMetadata Trusted app metadata of the service provider.
+     */
+    private static void clearTrustedAppCache(SpTrustedAppMetadata spTrustedAppMetadata) {
```

Comment:
Since we are not simply clearing the app cache but rather performing a validation and clearing the cache based on that lets rename the method to reflect that. Maybe something like ValidateAndClearTrustedAppCache